.github/workflows/CI.yaml

```diff
@@ -167,6 +167,52 @@ jobs:
         if: always()
         with:
           subfolder: 'unit-tests-backend'
+  python-tests:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'workflow_dispatch'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/logObjects
+        name: log job contexts
+        with:
+          env: ${{ toJSON(env) }}
+          job: ${{ toJSON(job) }}
+          steps: ${{ toJSON(steps) }}
+          strategy: ${{ toJSON(strategy) }}
+          matrix: ${{ toJSON(matrix) }}
+          needs: ${{ toJSON(needs) }}
+          inputs: ${{ toJSON(inputs) }}
+          vars: ${{ toJSON(vars) }}
+      - name: Prepare python environment
+        if: always()
+        run: | 
+          cd ./dataland-automated-qa-service
+          pip install --no-cache-dir -r requirements.txt
+          pip install -e .
+          openapi-python-client generate --path ../dataland-document-manager/documentManagerOpenApi.json
+          openapi-python-client generate --path ../dataland-backend/backendOpenApi.json
+          pip install --no-cache-dir ./dataland-backend-api-documentation-client
+          pip install --no-cache-dir ./dataland-document-manager-api-documentation-client
+      - name: Execute python unit tests and measure coverage
+        if: always()
+        run: |
+          cd ./dataland-automated-qa-service
+          pip install pytest
+          pip install pytest-cov
```

Comment:
TODO: check if pytest-cov is required or if we can use the standard coverage xml conversion